### ISSUE TITLE
AriaLiveAnnouncer: Assertion failed: value failed isValidValue: [object Object] on reset all

### ISSUE THREAD

**Issue by samreid** [phetsims/center-and-variability#113]
While working on https://github.com/phetsims/center-and-variability/issues/99, @chrisklus and I discovered this error when pressing reset all:

```
Assertion failed: value failed isValidValue: [object Object]
assert.js:25 Uncaught Error: Assertion failed: value failed isValidValue: [object Object]
    at window.assertions.assertFunction (assert.js:25:13)
    at Object.isValueValid (ValidatorDef.js? [sm]:324:39)
    at validate (validate.js? [sm]:36:18)
    at StateSchema.checkStateObjectValid (StateSchema.js? [sm]:185:9)
    at IOType.isStateObjectValid (IOType.js? [sm]:342:43)
    at IOType.validateStateObject (IOType.js? [sm]:384:10)
    at IOType.toStateObject (IOType.js? [sm]:223:24)
    at Object.toStateObject (NullableIO.js? [sm]:42:75)
    at IOType.toStateObject (IOType.js? [sm]:210:32)
    at Emitter.getPhetioData (PhetioDataHandler.ts:202:53)
```

**Comment by samreid** [comment]
We investigated this and there appears to be a problem in the state validator in OrIO.  It doesn't expect the `index` part of the state object.  We tried this and it had other problems:

```diff
Index: js/types/OrIO.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/types/OrIO.js b/js/types/OrIO.js
--- a/js/types/OrIO.js	(revision 264f9ebf1c92925159221614c509edbb1aea182e)
+++ b/js/types/OrIO.js	(date 1648512393115)
@@ -67,7 +67,11 @@
         return parameterTypes[ stateObject.index ].fromStateObject( stateObject.state );
       },
       stateSchema: StateSchema.asValue( `${typeNames.join( '|' )}`, {
-        isValidValue: isValidValue
+        isValidValue: stateObject => {
+          return isValidValue( stateObject.state ) &&
+                 stateObject.hasOwnProperty( 'index' ) &&
+                 typeof stateObject.index === 'number';
+        }
       } )
     } ) );
   }

```

@zepumph can you please advise?

**Comment by zepumph** [comment]
Yes thanks. There were two issues. The first was that I was causing a problem in Utterance. The second is that our error handling was incorrect for OrIO so the problem was hard to realize (hence your horrid message above).

Basically Utterance.toStateObject wasn't using OrIO to get the index/state object, so the state object was invalid. I also realized that the validator for OrIO was totally wrong, because it didn't use the same state object recursive checking that needed to occur at each level.

So pretty much https://github.com/phetsims/utterance-queue/commit/d85db291d0071e71f6663331b757ce8dcbf00035 fixed the bug, and https://github.com/phetsims/tandem/commit/fe151c20a974340d15410750f988668f18fb95d7 made it so that we could use OrIO on more complicated state objects like ChargedParticleIO which gets partial state from different levels in its IOType hierarchy.

Did this fix things for you as well?

**Comment by samreid** [comment]
Yes, it looks much better, thanks!

**Comment by samreid** [comment]
@chrisklus and I confirmed this is fixed in in https://github.com/phetsims/center-and-variability/issues/122, closing.
